### PR TITLE
Feat: admit outputs with op_return script in utxo chains

### PR DESCRIPTION
**Pull request by Gamboster**

### CODE REVIEW

File: packages/bitcore-wallet-service/test/integration/server.js [hunk 1]
```diff
@@ -3969,8 +3969,67 @@ describe('Wallet service', function() {
               });
             });
           });
-          if(coin === 'btc' || coin === 'bch' || coin === 'doge' || coin === 'ltc') {
-            it('should fail to create BTC/BCH tx for invalid amount', function(done) {
```
Comment (Amirshahnavazi64) [inline packages/bitcore-wallet-service/test/integration/server.js:3973]: 18R3a81j23oqtSTTJxwXXJ3waaWbrY3KRv

**Comment by Amirshahnavazi64** [comment]
18R3a81j23oqtSTTJxwXXJ3waaWbrY3KRv

**Comment by Amirshahnavazi64** [comment]
18R3a81j23oqtSTTJxwXXJ3waaWbrY3KRv

File: packages/bitcore-wallet-service/src/lib/chain/btc/index.ts [hunk 2]
```diff
@@ -450,8 +463,18 @@ export class BtcChain implements IChain {
     const serializationOpts = {
       disableIsFullySigned: true,
       disableSmallFees: true,
-      disableLargeFees: true
+      disableLargeFees: true,
+      disableDustOutputs: false
     };
+
+    if (txp.outputs && _.isArray(txp.outputs)) {
```
Comment (kajoseph) [inline packages/bitcore-wallet-service/src/lib/chain/btc/index.ts:470]: Array.isArray

File: packages/bitcore-wallet-service/src/lib/chain/btc/index.ts [hunk 3]
```diff
@@ -210,6 +210,19 @@ export class BtcChain implements IChain {
...
+      if (typeof output.script !== 'string') {
+        return Errors.SCRIPT_TYPE;
+      }
+
+      // check OP_RETURN
+      if (output.script.startsWith('6a') && output.amount != 0) {
+        return Errors.SCRIPT_OP_RETURN_AMOUNT;
+      }
+    }
+  }
```
Comment (kajoseph) [inline packages/bitcore-wallet-service/src/lib/chain/btc/index.ts:224]: This is blank check for a non OP_RETURN script. Perhaps to protect against potential script attacks, we should _only_ support OP_RETURN scripts for now.

**Review by kajoseph — CHANGES_REQUESTED** [review]

File: packages/bitcore-wallet-service/src/lib/chain/btc/index.ts (hunk 3 above)
Comment (Gamboster) [reply packages/bitcore-wallet-service/src/lib/chain/btc/index.ts:229]: Done. I added a check to ensure that script is only of type OP_RETURN

File: packages/bitcore-wallet-service/src/lib/chain/btc/index.ts (hunk 2 above)
Comment (Gamboster) [reply packages/bitcore-wallet-service/src/lib/chain/btc/index.ts]: Done.

**Review by kajoseph — APPROVED** [review]
tACK. Nice work!

**Pull request merged**